node/src/exchange.rs

```diff
@@ -41,11 +44,49 @@ impl Codec for HeaderCodec {
     where
         T: AsyncRead + Unpin + Send,
     {
-        let mut vec = Vec::new();
+        let mut buf = bytes::BytesMut::with_capacity(512);
+        buf.resize(PROTOBUF_MAX_LENGTH_DELIMITER_LEN, 0);
+
+        let mut read = 0;
+        let len = loop {
+            match io.read(&mut buf[read..]).await? {
+                0 => {
+                    return Err(io::Error::new(
+                        io::ErrorKind::UnexpectedEof,
+                        ReadHeaderError::StreamClosed,
+                    ));
+                }
+                n => read += n,
+            };
+
+            if let Ok(len) = prost::decode_length_delimiter(&buf[..read]) {
+                break len;
+            }
+        };
+
+        // const value conversion, safe for 32bit and larger usize
+        if len
+            > REQUEST_SIZE_MAXIMUM
```

Comment:
wouldn't it be better to just define those constants as usize?

Reply:
We're the only users of this const, so actually yes. Changed :+1: